src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -274,6 +278,22 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
             _workerInitTask.SetResult(true);
         }
 
+        private void ParseAndLogWorkerMetadata()
+        {
+            if (_initMessage?.WorkerMetadata == null)
+            {
+                return;
+            }
+
+            _initMessage.WorkerMetadata.RuntimeName = string.IsNullOrEmpty(_initMessage.WorkerMetadata.RuntimeName)
+                                                        ? _workerConfig.Description.Language : _initMessage.WorkerMetadata.RuntimeName;
+            _initMessage.WorkerMetadata.RuntimeVersion = string.IsNullOrEmpty(_initMessage.WorkerMetadata.RuntimeVersion)
+                                                        ? _workerConfig.Description.DefaultRuntimeVersion : _initMessage.WorkerMetadata.RuntimeVersion;
+
+            _metricsLogger.LogEvent(MetricEventNames.WorkerMetadata, null, _initMessage.WorkerMetadata.ToString());
```

Comment:
nit: Consider using the argument name to help with readability for the `null` param.